dot/rpc/modules/childstate_integration_test.go

```diff
@@ -0,0 +1,298 @@
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+package modules
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/ChainSafe/chaindb"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/trie"
+	"github.com/stretchr/testify/require"
+)
+
+func TestChildStateGetKeys(t *testing.T) {
+	childStateModule, currBlockHash := setupChildStateStorage(t)
+
+	req := &GetKeysRequest{
+		Key:    []byte(":child_storage_key"),
+		Prefix: []byte{},
+		Hash:   nil,
+	}
+
+	res := make([]string, 0)
+	err := childStateModule.GetKeys(nil, req, &res)
+	require.NoError(t, err)
+	require.Len(t, res, 3)
+
+	for _, r := range res {
+		b, dErr := common.HexToBytes(r)
+		require.NoError(t, dErr)
+		require.Contains(t, []string{
+			":child_first", ":child_second", ":another_child",
+		}, string(b))
+	}
+
+	req = &GetKeysRequest{
+		Key:    []byte(":child_storage_key"),
+		Prefix: []byte(":child_"),
+		Hash:   &currBlockHash,
+	}
+
+	err = childStateModule.GetKeys(nil, req, &res)
+	require.NoError(t, err)
+	require.Len(t, res, 2)
+
+	for _, r := range res {
+		b, err := common.HexToBytes(r)
+		require.NoError(t, err)
+		require.Contains(t, []string{
+			":child_first", ":child_second",
+		}, string(b))
+	}
+}
+
+func TestChildStateGetStorageSize(t *testing.T) {
+	mod, blockHash := setupChildStateStorage(t)
+	invalidHash := common.BytesToHash([]byte("invalid block hash"))
+
+	tests := []struct {
+		expect   uint64
+		err      error
+		hash     *common.Hash
+		keyChild []byte
+		entry    []byte
+	}{
+		{
+			err:      nil,
+			expect:   uint64(len([]byte(":child_first_value"))),
+			hash:     nil,
+			entry:    []byte(":child_first"),
+			keyChild: []byte(":child_storage_key"),
+		},
+		{
+			err:      nil,
+			expect:   uint64(len([]byte(":child_second_value"))),
+			hash:     &blockHash,
+			entry:    []byte(":child_second"),
+			keyChild: []byte(":child_storage_key"),
+		},
+		{
+			err:      nil,
+			expect:   0,
+			hash:     nil,
+			entry:    []byte(":not_found_so_size_0"),
+			keyChild: []byte(":child_storage_key"),
+		},
+		{
+			err:      fmt.Errorf("child trie does not exist at key %s%s", trie.ChildStorageKeyPrefix, []byte(":not_exist")),
+			hash:     &blockHash,
+			entry:    []byte(":child_second"),
+			keyChild: []byte(":not_exist"),
+		},
+		{
+			err:  chaindb.ErrKeyNotFound,
+			hash: &invalidHash,
+		},
+	}
+
+	for _, test := range tests {
+		var req GetChildStorageRequest
+		var res uint64
+
+		req.Hash = test.hash
+		req.EntryKey = test.entry
+		req.KeyChild = test.keyChild
```

Comment:
```suggestion
		req := GetChildStorageRequest{
			Hash: test.hash,
			EntryKey: test.entry,
			KeyChild: test.keyChild,
		}
		var res uint64

```

also why not have `var req GetChildStorageRequest` directly as a field in your test cases?